pkg/controller/common.go

```diff
@@ -778,34 +675,226 @@ func (s *scoreByReplicaCount) name() string {
 	return "score by replica count"
 }
 
-func (s *scoreByReplicaCount) setup(pods []v1.Pod, volumes []string, state *SharedState) {
+func (s *scoreByReplicaCount) priority() float64 {
+	return 3
+}
+
+func (s *scoreByReplicaCount) setup(nodes []v1.Node, pods []v1.Pod, volumes []string, state *SharedState) {
 	s.volumes = volumes
 	s.state = state
 }
 
 func (s *scoreByReplicaCount) score(ctx context.Context, nodeScores map[string]int) (map[string]int, error) {
 	ctx, w := workflow.New(ctx, workflow.WithDetails("score-plugin", s.name()))
 	defer w.Finish(nil)
+
+	var requestedReplicaCount int
+
+	nodeReplicaCounts := map[string]int{}
+
 	for _, volume := range s.volumes {
+		azVolume, err := azureutils.GetAzVolume(ctx, s.state.cachedClient, nil, volume, s.state.objectNamespace, true)
+		if err != nil {
+			w.Logger().V(5).Errorf(err, "failed to get AzVolume (%s)", volume)
```

Comment:
Errors should probably be logged at the base level rather than as verbose since they would have externally visible effects.